src/sound/songset.cc

```diff
@@ -18,24 +18,71 @@
 
 #include "sound/songset.h"
 
-#include <cassert>
+#include <SDL_mixer.h>
 
 #include "base/log.h"
 #include "io/fileread.h"
 #include "io/filesystem/layered_filesystem.h"
+#include "wlapplication_options.h"
 
 /// Prepare infrastructure for reading song files from disk and register the matching files
 Songset::Songset(const std::string& dir, const std::string& basename) {
 	assert(g_fs);
-	std::vector<std::string> mp3_files = g_fs->get_sequential_files(dir, basename, "mp3");
-	std::vector<std::string> ogg_files = g_fs->get_sequential_files(dir, basename, "ogg");
-	add_songs(mp3_files);
-	add_songs(ogg_files);
+
+	name_ = basename;
+
+	std::vector<std::string> mp3_files = g_fs->get_sequential_files(dir, name_, "mp3");
+	std::vector<std::string> ogg_files = g_fs->get_sequential_files(dir, name_, "ogg");
+
+	load_songs();
+
+	if (songs_.empty()) {
+		init_songs(mp3_files);
+		init_songs(ogg_files);
+		load_songs();
+	}
+
+	// initialize "neg" index to ensure 1st sequential playback starts at 0
+	// in case you wonder, yes this is needed
+	current_song_ = static_cast<uint32_t>(-1);
 }
-void Songset::add_songs(const std::vector<std::string>& files) {
+
+/**
+ * Initializes config from a list of music files, enabled by default
+ * \param files filenames that will be added
+ */
+void Songset::init_songs(std::vector<std::string> files) {
 	for (const std::string& filename : files) {
-		assert(!g_fs->is_directory(filename));
-		add_song(filename);
+		set_config_bool("songs", filename, true);
+	}
+}
+
+/// Loads song data from config into memory
+void Songset::load_songs() {
+	const std::string& path_basename = "music/" + name_;
```

Comment:
```suggestion
	std::string& path_basename = "music";
	path_basename += FileSystem::file_separator() + name_;
```

this worked for me to fix this.  I agree with @cb/Nordfriese that only fixing this is crucial as it prevents any music in windows at all currently